### ISSUE TITLE
Nulling Audio Clip Field Doesn't Always Clear Audio

### ISSUE THREAD

**Issue by Enverex** [Frooxius/NeosPublic#1431]
I'm running into an issue where I need to switch audio sources a lot of on the fly and it seems that Neos for some reason doesn't always unload the audio clip when the URL is nulled. As can be seen by this image, the audio data is still there (with the duration, etc) and playing, despite the fact it should be empty/removed.

![image](https://user-images.githubusercontent.com/958670/101084380-c7080680-35a5-11eb-9b4e-6f133edfafb0.png)

**Comment by Frooxius** [comment]
I can't seem to replicate this, it nulls it correctly on my end. There was a bug like this a while back in general with references, but that's been since fixed. Are you still experiencing this on the latest build?

**Comment by Enverex** [comment]
Just had this happen in another world. One of the cats was set to null yet started playing one of the other songs (either that I had open in another world, or had been used earlier, can't tell which unfortunately).

**Comment by Frooxius** [comment]
Is there an easy way to replicate it with that setup?

**Comment by Enverex** [comment]
The easiest way would probably be as follows: Go to a world you have builder in, a new "Empty World" would be fine. Play a song that has an instrument in it that isn't that common but easy to find, e.g. an Acoustic Guitar. Wait for that to actually play. Now switch to another song that doesn't use that instrument. Logix will have cleared all instruments and then loaded new tracks into the cats that are used for the new song. Now play that. Watch the instrument that shouldn't be used, eventually it will also start playing the old song again once it reaches the part where that audio track has audio (you can inspect the cat to see that there's no URL for the clip anymore - there will be a Audio Clip Provider on the Head Proxy).

I'll send you my public folder with it in.

**Comment by Enverex** [comment]
It looks like it's even more random than I thought. Here's me loading into a world with completely blank dynamic variables and nothing already set. I then load one song. Then load a second. As you can see the selected cat has a null URL in both cases, yet at the same time it clearly shows (and plays) a loaded audio clip. What wasn't shown in this video is one of the other cats started playing a track from a completely different song (not one of the two I clicked) yet again also had a null clip URL. It's almost like it's pulling random things out of the cache somehow.

https://cdn.discordapp.com/attachments/472301664271859723/785236032827818011/2020-12-06_20-02-49.mp4

**Comment by Frooxius** [comment]
Thanks, I was able to replicate it with this. It was a more general error with Static asset providers (meaning same would happen to textures and meshes as well), where partially loaded asset would finish loading, even if the URL was set to Null in the meanwhile.

It should be fixed in 2020.12.8.638, let me know if there are still trouble with it, but I did a fair bit of testing with the cats and it didn't seem to occur again.
